### PR TITLE
Support Sentinel with TLS

### LINKED ISSUE #1987: Support TLS connections in Sentinel mode
## Feature Request

Redis open source will [support TLS](https://github.com/antirez/redis/pull/4855) in the not distant future.  TLS support in the Sentinel will likely follow soon thereafter.

Please implement TLS support in Jedis to support Sentinel mode.  This would include not only connecting to the Sentinel over TLS, but also connecting to the sentinel-reported master over TLS.

#### Is your feature request related to a problem? Please describe

Financial services and enterprise users of Jedis, including those using Jedis by way of Spring Data Redis, are increasingly being asked to ensure all service traffic is encrypted via TLS.  

#### Describe alternatives you've considered

* removing calls to the Sentinel API in our code, migrating to the lower level Redis API for Redis Cluster (when it supports TLS - see link above) or Redis Enterprise.  This would be painful for the number of applications we have that program to the Sentinel API.
* stunnel and spiped, neither of which work well for how our services are deployed and maintained
* exotic IPSec solutions to encrypt traffic on the wire in lieu of first class TLS support in Sentinel mode

### PR DESCRIPTION
**Pull request by sazzad16**
Original author: @Talon876 (#2139)

Resolves #1987 

Closes #2024 
Closes #2139 
Closes #2232

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -243,6 +243,12 @@ pid = /tmp/stunnel.pid
 [redis]
 accept = 127.0.0.1:6390
 connect = 127.0.0.1:6379
+[redis_3]
```
Comment (gkorland) [inline Makefile:246]: @sazzad16 where is redis_2?

Comment (sazzad16) [reply Makefile:246]: @gkorland This piece of code is from #2139 . I think the OP left redis_2 for the connection in 6380 port.

**Comment by vkravets** [comment]
Any updates about this PR? We want to use it, but it still haven't released yet (

**Comment by sazzad16** [comment]
Let's check #2403 

@gkorland ping!

**Pull request closed without merging**